serde_v8/src/magic/value.rs

```diff
@@ -29,50 +27,22 @@ impl<'s> From<Value<'s>> for v8::Local<'s, v8::Value> {
   }
 }
 
-impl serde::Serialize for Value<'_> {
-  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-  where
-    S: serde::Serializer,
-  {
-    use serde::ser::SerializeStruct;
-
-    let mut s = serializer.serialize_struct(NAME, 1)?;
-    let mv = Value {
-      v8_value: self.v8_value,
-    };
-    let hack: u64 = unsafe { std::mem::transmute(mv) };
-    s.serialize_field(FIELD, &hack)?;
-    s.end()
+impl ToV8 for Value<'_> {
+  fn to_v8<'a>(
+    &self,
+    _scope: &mut v8::HandleScope<'a>,
+  ) -> Result<v8::Local<'a, v8::Value>, crate::Error> {
+    // SAFETY: not fully safe, since lifetimes are detached from original scope
+    Ok(unsafe { transmute(self.v8_value) })
   }
 }
 
-impl<'de, 's> serde::Deserialize<'de> for Value<'s> {
-  fn deserialize<D>(deserializer: D) -> Result<Value<'s>, D::Error>
-  where
-    D: serde::Deserializer<'de>,
-  {
-    struct ValueVisitor<'s> {
-      p1: PhantomData<&'s ()>,
-    }
-
-    impl<'de, 's> serde::de::Visitor<'de> for ValueVisitor<'s> {
-      type Value = Value<'s>;
-
-      fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
-        formatter.write_str("a v8::Value")
-      }
-
-      fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
-      where
-        E: serde::de::Error,
-      {
-        let mv: Value<'s> = unsafe { std::mem::transmute(v) };
-        Ok(mv)
-      }
-    }
-
-    static FIELDS: [&str; 1] = [FIELD];
-    let visitor = ValueVisitor { p1: PhantomData };
-    deserializer.deserialize_struct(NAME, &FIELDS, visitor)
+impl FromV8 for Value<'_> {
+  fn from_v8(
+    _scope: &mut v8::HandleScope,
+    value: v8::Local<v8::Value>,
+  ) -> Result<Self, crate::Error> {
+    // SAFETY: not fully safe, since lifetimes are detached from original scope
```

Comment:
Ditto

Reply:
This is never directly called by end-users. It's the result of attempting to de/serialize serde_v8::Values